eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/LegacyJpaEventStorageEngine.java

```diff
@@ -0,0 +1,653 @@
+            }
+        }
+
+        public static Customization withDefaultValues() {
+            return new Customization(
+                    NoOpEventUpcaster.INSTANCE,
+                    null,
+                    DEFAULT_BATCH_SIZE,
+                    null,
+                    new ClassBasedMessageNameResolver(),
+                    DEFAULT_EXPLICIT_FLUSH,
+                    DEFAULT_LOWEST_GLOBAL_SEQUENCE,
+                    TokenGapsHandlingConfig.withDefaultValues()
+            );
+        }
+
+        public Customization upcasterChain(EventUpcaster upcasterChain) {
+            return new Customization(upcasterChain,
+                                     persistenceExceptionResolver,
+                                     batchSize,
+                                     finalAggregateBatchPredicate,
+                                     messageNameResolver,
+                                     explicitFlush,
+                                     lowestGlobalSequence,
+                                     tokenGapsHandling
+            );
+        }
+
+        public Customization persistenceExceptionResolver(PersistenceExceptionResolver persistenceExceptionResolver) {
+            return new Customization(upcasterChain,
+                                     persistenceExceptionResolver,
+                                     batchSize,
+                                     finalAggregateBatchPredicate,
+                                     messageNameResolver,
+                                     explicitFlush,
+                                     lowestGlobalSequence,
+                                     tokenGapsHandling
+            );
+        }
+
+        public Customization batchSize(int batchSize) {
+            return new Customization(upcasterChain,
+                                     persistenceExceptionResolver,
+                                     batchSize,
+                                     finalAggregateBatchPredicate,
+                                     messageNameResolver,
+                                     explicitFlush,
+                                     lowestGlobalSequence,
+                                     tokenGapsHandling
+            );
+        }
+
+        public Customization finalAggregateBatchPredicate(
+                Predicate<List<? extends DomainEventData<?>>> finalAggregateBatchPredicate) {
+            return new Customization(upcasterChain,
+                                     persistenceExceptionResolver,
+                                     batchSize,
+                                     finalAggregateBatchPredicate,
+                                     messageNameResolver,
+                                     explicitFlush,
+                                     lowestGlobalSequence,
+                                     tokenGapsHandling
+            );
+        }
+
+        public Customization messageNameResolver(MessageNameResolver messageNameResolver) {
```

Comment:
The MessageName shouldn't be resolved at this point. This is only relevant when a new EventMessage is constructed from an arbitrary payload object. Typically in Gateways

Reply:
Ah OK, so in this case, the name should be stored and read from the store afterwards.